### PR TITLE
adding GoogleAuth, change API to local JSON API

### PR DESCRIPTION
**Pull request by BeataKula**
adding GoogleAuth, change API to locaj JSON API

### CODE REVIEW

File: client/src/components/GoogleAuth.js [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+            window.gapi.client
+                .init({
+                    clientId:
+                        "923811828877-90t74vf4k8tpms3gvivbogqmrafp9t7r.apps.googleusercontent.com",
+                    scope: "email",
+                })
+                .then(() => {
+                    authInstance = window.gapi.auth2.getAuthInstance();
+                    onAuthChange(authInstance.isSignedIn.get());
+                    // Ustawienie nasłuchu w przypadku zmiany statusu zalogowania usera
```
Comment (zgr92) [inline client/src/components/GoogleAuth.js:39]: Always a good practice - comments and code only in English

File: client/src/components/GoogleAuth.js [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    } else if (isSignedIn) {
+        return (
+            <div
+                style={{
+                    width: "100%",
+                    textAlign: "right",
+                }}
+            >
+                <Button
+                    style={{
```
Comment (zgr92) [inline client/src/components/GoogleAuth.js:65]: Why in inline style? Why not styled component?

File: client/src/components/GoogleAuth.js [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+                    color="default"
+                    startIcon={<ExitToAppIcon />}
+                    onClick={onSignOutClick}
+                    id="sign-out-button"
+                >
+                    Sign Out
+                </Button>
+            </div>
+        );
+    } else {
```
Comment (zgr92) [inline client/src/components/GoogleAuth.js:79]: It's a matter of style and preference, but personally I find if statements without multiple elses more readable, especially when you already have return statements in each of the ifs. Something like mentioned here - https://janalmazora.medium.com/how-to-prevent-using-if-else-statements-in-your-code-7e05e43afde

File: client/src/components/Post.tsx [hunk 4]
```diff
@@ -66,25 +44,15 @@ const Post: React.FunctionComponent<PostProps> = ({
...
-                        onClick={addComment}
-                    />
-                </PostElementFooterStyle>
-            </PostElementStyle>
-        </li>
+        <>
+            <PostAuthor>
+                <UserHeader user={user} userId={userId} />
+            </PostAuthor>
+            <PostTitle>
```
Comment (zgr92) [inline client/src/components/Post.tsx:51]: Could be a separate component with `<b> `inside and only `<PostTitle>{title}</PostTitle>` here

File: client/src/pages/PostList.tsx [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+import Message from "../components/Message";
+import Button from "../components/Button";
+import { StyledButtonLink } from "../components/LinkComponent";
+import { PostProps, ReducersState } from "../AppTypes";
+import Comment from "../components/Comment";
+import { bindActionCreators, Dispatch } from "redux";
+
+const ADD_BLOG_LINK = "/Blog/add";
+
+export const PostsListStyle = styled.ul`
```
Comment (zgr92) [inline client/src/pages/PostList.tsx:16]: It's just PostList element, isn't it? It's not just a style, it's a component, so no need to add "style" in the name I think.

File: client/src/pages/PostList.tsx [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+    };
+
+    componentDidMount() {
+        this.props.fetchPostsAndUsers();
+    }
+
+    componentDidUpdate() {
+        const allList = this.props.postsReducer.allList;
+
+        if (
```
Comment (zgr92) [inline client/src/pages/PostList.tsx:55]: Is it possible to simplify those conditions?

**Review by zgr92 — APPROVED** [review]

File: client/src/pages/PostList.tsx (hunk 5 above)
Comment (BeataKula) [reply client/src/pages/PostList.tsx:16]: In this PostsListStyle there are only styles for list
 padding: 15px;
    padding-top: 0px;
    list-style-type: none;
    z-index: 10;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(400px, 1fr));
    grid-gap: 10px;

**Pull request merged**